src/services/registrationData.ts

```diff
@@ -74,3 +74,81 @@ export async function overwriteRegistrationData({
   // return new registration data
   return newRegistrationData;
 }
+
+export async function updateQuestionOptions(
+  // inserts new question options or updates existing ones
+  dbPool: PostgresJsDatabase<typeof db>,
+  registrationData:
+    | {
+        id: string;
+        registrationFieldId: string;
+        value: string;
+      }[]
+    | null,
+): Promise<void> {
+  try {
+    if (!registrationData) {
+      return;
+    }
+
+    // Fetch registration_field_ids from registrationData that have
+    // a question_id so it requires question_options to be populated
+    const registrationFields = await dbPool.execute<{
+      registrationFieldId: string;
+      questionId: string;
+    }>(
+      sql.raw(`
+          SELECT id AS "registrationFieldId", question_id AS "questionId"
+          FROM registration_fields
+          WHERE question_id IS NOT NULL
```

Comment:
this can be optimized by only getting the registrationFields that are sent in registrationData array

Reply:
optimized query by including an additional where statement: `AND id IN (${registrationData.map((data) => `'${data.registrationFieldId}'`).join(', ')})`